service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -447,6 +447,50 @@ public ReportWithLibrary createReportWithLibrary(
         newReport.getParentSupport());
   }
 
+  @Override
+  public void adminUpdateReportWithLibrary(
+      JWTData userData, int schoolId, int reportId, UpsertReportWithLibrary req) {
+
+    if (!userData.isAdmin()) {
+      throw new AdminOnlyRouteException();
+    }
+    SchoolsRecord school = this.queryForSchool(schoolId);
+    if (school == null) {
+      throw new SchoolDoesNotExistException(schoolId);
+    }
+
+    // Save a record to the school_reports_with_libraries table
+    SchoolReportsWithLibrariesRecord newReport = db.selectFrom(SCHOOL_REPORTS_WITH_LIBRARIES)
+        .where(SCHOOL_REPORTS_WITH_LIBRARIES.ID.eq(reportId)).fetchOne();
+
+    if (newReport == null) {
+      throw new NoReportFoundException(schoolId);
+    }
+
+    newReport.setUserId(userData.getUserId());
```

Comment:
🌱 I wonder if there's a way we can just automatically bind these parameters to the request, as opposed to having to explicitly set each one